spec/requests/requests_spec.rb

```diff
@@ -160,25 +420,188 @@
+          subject.call
+          expect(page).to have_content('Super FastReplier')
+          expect(page).to have_content('First message')
+
+          expect(page).to have_content('Second AintBad')
+          expect(page).to have_content('Second message')
+
+          expect(page).not_to have_content('Not part of request')
+          expect(page).not_to have_content('Not part of organization')
+        end
+      end
+    end
+  end
+
+  describe 'GET /:organization_id/requests/:id/stats' do
+    subject { -> { get stats_organization_request_path(organization, request, as: user) } }
+
+    let(:request) { create(:request, organization: organization) }
+
+    context 'unauthenticated' do
+      let(:user) { nil }
+
+      it 'redirects to the sign in path' do
+        subject.call
+        expect(response).to redirect_to(sign_in_path)
+      end
+    end
+
+    context 'unauthorized' do
+      let(:user) { create(:user, organizations: [create(:organization)]) }
+
+      it 'renders not found ' do
+        subject.call
+        expect(response).to be_not_found
+      end
+    end
+
+    context 'authenticated and authorized' do
+      let(:user) { create(:user, organizations: [organization]) }
+
+      context 'request not part of organization' do
+        let(:request) { create(:request, organization: create(:organization)) }
+
+        it 'renders not found ' do
+          subject.call
+          expect(response).to be_not_found
+        end
+      end
+
+      context 'request part of organization' do
+        describe 'given a number of requests, replies and photos' do
+          before(:each) do
+            create_list(:message, 2)
+            delivered_messages = create_list(:message, 7, :outbound, request: request, broadcasted: true)
+            create(:message, :with_file, :outbound, request: request, broadcasted: false,
+                                                    attachment: fixture_file_upload('example-image.png'))
+            # _ is some unresponsive recipient
+            responsive_recipient, _, *other_recipients = delivered_messages.map(&:recipient)
+            create_list(:message, 3, request: request, sender: responsive_recipient)
+            other_recipients.each do |recipient|
+              create(:message, :with_a_photo, sender: recipient, request: request)
+              create(:message, :with_file, sender: recipient, request: request, attachment: fixture_file_upload('example-image.png'))
+              create(:message, :with_file, sender: recipient, request: request,
+                                           attachment: fixture_file_upload('invalid_profile_picture.pdf'))
+            end
+            request.reload
+          end
+
+          it 'displays the stats' do
+            subject.call
+            expect(page).to have_content('6/8')
+            expect(page).to have_content('18')
+            expect(page).to have_content('10')
+          end
+        end
```

Comment:
I'm not sure about this test. This is a lot of overhead to check for these numbers. What do you think @soey 
The only reason this is a separate action in the controller is because we lazy load these stats to improve the time it takes to render the requests index page. These stats calculations are quite expensive so we moved them to be calculated on demand.

Reply:
I would say: If it's not tested it's probably broken. So I think testing is justified. My problem with that is that it's hard to understand how the numbers refer to the test setup. I would probably add more context to it. If you feel there is too much logic testing on the request then it could be justified to extract a data object into a PORO and test it encapsulated in a model spec.

Reply:
Well, the request stats are tested in https://github.com/tactilenews/100eyes/blob/main/spec/models/request_spec.rb#L204-L275

The setup is copy/pasted from there. I agree it's not super intuitive. This is also only testing it at a unit level and is slightly different because here we test that the inline metrics display the "contributors" who replied of those that received as a division, ie `6/8` and is end-to-end.

We could simplify it to send the message to two recipients, one of which responded several times, one with a photo.